blueocean-dashboard/src/main/js/components/karaoke/components/Pipeline.jsx

```diff
@@ -0,0 +1,272 @@
+            case 'pipeline_block_end':
+            case 'pipeline_stage': {
+                logger.debug('sse event block starts refetchNodes', jenkinsEvent);
+                debounce(() => {
+                    logger.debug('sse fetch it', this.karaoke);
+                    this.pager.fetchNodes({});
+                }, 200)();
+                // prevent flashing of stages and nodes
+                this.showPending = false;
+                break;
+            }
+            default: {
+                logger.debug('ignoring event', jenkinsEvent);
+            }
+            }
+        } catch (e) {
+            // we only ignore the exit error
+            if (e.message !== 'exit') {
+                logger.error('sse Event has produced an error, will not work as expected.', e);
+            }
+        }
+    }
+
+    render() {
+        const { t, run, augmenter, branch, pipeline, router, scrollToBottom, location } = this.props;
+        // do we have something to display?
+        const noResultsToDisplay = this.pager.steps === undefined || (this.pager.steps && !this.pager.steps.data.hasResultsForSteps);
+        // Queue magic since a pipeline is only showing queued state a short time even if still waiting for executors
+        const isPipelineQueued = run.isQueued() || (run.isRunning() && noResultsToDisplay);
+        logger.debug('isQueued', run.isQueued(), 'noResultsToDisplay', noResultsToDisplay, 'isPipelineQueued', isPipelineQueued);
+        const queuedMessage = t('rundetail.pipeline.queued.message', { defaultValue: 'Waiting for run to start' });
+        if (run.isQueued()) { // if queued we are saying that we are waiting to start
+            logger.debug('EarlyOut - abort due to run queued.');
+            return <QueuedState message={queuedMessage} />;
+        }
+        const supportsNodes = this.pager.nodes === undefined;
+        if(!this.pager.pending && (this.classicLog  || ( noResultsToDisplay && supportsNodes))) { // no information? fallback to freeStyle
+            logger.debug('EarlyOut - We do not have any information we can display or we opt-out by preference, falling back to freeStyle rendering');
+            return (<FreeStyle {...this.props }/>);
+        }
+        if (this.pager.pending && this.showPending) { // we are waiting for the backend information
+            logger.debug('EarlyOut - abort due to pager pending');
+            const pendingMessage = t('rundetail.pipeline.pending.message', { defaultValue: 'Waiting for backend to response' });
+            return <QueuedState message={pendingMessage} />;
+        }
+        // here we decide what to do next if somebody clicks on a flowNode
+        // Underlying tasks are fetching nodes information for the selected node
+        const afterClick = (id) => {
+            logger.debug('clicked on node with id:', id);
+            this.showPending = false; // Configure flag to not show pending anymore -> reduce flicker
```

Comment:
Since if pending is activated in general, we do not want to show pending when clicking on finished nodes. You can see pending when you set `runDetails.pipeline.showPending`  to `always` in the localStorage.